src/main/java/org/apache/pirk/schema/query/QuerySchemaLoader.java

```diff
@@ -246,8 +231,11 @@ private static QuerySchema loadQuerySchemaFile(String schemaFile, boolean hdfs,
     }
 
     // Create the query schema object
-    querySchema = new QuerySchema(schemaName, dataSchemaName, elementNames, selectorName, dataElementSize, filterNamesSet, filter);
 
+    DataFilter filter = instantiateFilter(filterTypeName, filterNamesSet);
+    querySchema = new QuerySchema(schemaName, dataSchemaName, selectorName, filterTypeName, filter, dataElementSize);
```

Comment:
As a very small (kind of nitty) style item, I would much rather have setter methods for QuerySchema to set the `elementName`  and `filterNamesSet` rather than setting these fields via 

```
querySchema.getElementNames().addAll(elementNames);
querySchema.getFilteredElementNames().addAll(filterNamesSet);
```

In the original code, these were passed in as part of the constructor (which has now been removed in this refactor).